publish-externals/src/index.ts

```diff
@@ -10,28 +12,89 @@ const ExternalPublications = [
 
 const main = async () => {
   const changedFiles = process.argv[2].split(" ")
-  const errors = []
+  const blogs = snapshot.snapshot || []
+
+  // publish changed files
   for (const file of changedFiles) {
     if (file.startsWith("blog/")) {
-      const filePath = path.join(__dirname, "../../", file)
-      const {frontMatter, content} = extractFrontMatterAndContent(filePath)
-      for (let publication of ExternalPublications) {
-        console.log(`[${publication.name}] ${frontMatter.slug} ... publishing ⏳`)
-        try {
-          await publication.handler(frontMatter, content)
-          console.log(`[${publication.name}] Success ${frontMatter.slug} ... succeeded ✅`)
-        } catch (error) {
-          errors.push(error)
-          console.error(`[${publication.name}] Failure ${frontMatter.slug} ... failed 💀`)
+      await publish(file.replace("blog/", ""))
+    }
+  }
+
+  // publish blogs/updates which were not published before
+  for (const blog of blogs) {
+    if (!blog.lastUpdatePublished || !blog.published) {
+      await publish(blog.blogName)
+    }
+  }
+}
+
+const publish = async (file: string) => {
+  const blogs = snapshot.snapshot || []
+  const errors = []
+  const filePath = path.join(__dirname, "../../", `blog/${file}`)
+  const {frontMatter, content} = extractFrontMatterAndContent(filePath)
+  for (let publication of ExternalPublications) {
+    console.log(`[${publication.name}] ${frontMatter.slug} ... publishing ⏳`)
+    try {
+      await publication.handler(frontMatter, content)
```

Comment:
Publish if either of them is true:
1. The blog isn't already published — check from snapshots
2. Hash of the blog has changed.